### PR TITLE
Add codespell: config, workflow, pre-commit config and make it fix few typos

### PR DESCRIPTION
**Pull request by yarikoptic**
- [ ] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

it doesn't so I didn't bother

I see that once in a while commits fixing typos come in. With https://github.com/codespell-project/codespell you could minimize them even further since it would be harder for them to sneak in.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -86,3 +93,5 @@ package = "pipx"
 [[tool.mypy.overrides]]
 module = ["pycowsay.*"]
 ignore_missing_imports = true
+
+# Ref: https://github.com/codespell-project/codespell#using-a-config-file
```
Comment (yarikoptic) [inline pyproject.toml:97]: I guess I better just remove that comment since file could be auto resorted and comments disappear

```suggestion
```

**Comment by chrysle** [comment]
Thanks! Why did you add `codespell` both as a pre-commit hook and a GH Actions workflow?

**Comment by yarikoptic** [comment]
just what my https://github.com/yarikoptic/improveit/blob/main/codespellit does since not necessarily people have pre-commit as a service or a workflow ;) I will drop the commit adding github action now.

**Comment by chrysle** [comment]
LGTM, please rebase or merge.

File: pyproject.toml [hunk 2]
```diff
@@ -71,6 +71,14 @@ lint.extend-select = [
 lint.isort = {known-first-party = ["helpers", "package_info", "pipx"]}
 lint.mccabe.max-complexity = 15
 
+[tool.codespell]
+# Ref: https://github.com/codespell-project/codespell#using-a-config-file
+skip = '.git,*.pdf,*.svg,go.sum'
```
Comment (Gitznik) [inline pyproject.toml:76]:
```suggestion
skip = '.git,*.pdf,*.svg'
```
I don't think we will have a `go.sum` in this project.

**Review by Gitznik — CHANGES_REQUESTED** [review]
Thanks for suggesting this. Let's hope it will cut some manual effort in the future.
Have you run it on all current files?

File: pyproject.toml (hunk 2 above)
Comment (yarikoptic) [reply pyproject.toml:76]: will remove then

**Comment by yarikoptic** [comment]
> Thanks for suggesting this. Let's hope it will cut some manual effort in the future. Have you run it on all current files?

yes, codespell considers all the applicable (non binary) files AFAIK. There is a commit done via `datalad run` with annotation on execution

> Also - is it necessary to exclude the vens nox creates from checking when doing a run on all files?

cool, adding `.nox` to excludes (and something for my `codespellit` script to look for)

**Comment by yarikoptic** [comment]
> LGTM, please rebase or merge.

pushed rebased

**Review by Gitznik — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -71,6 +71,14 @@ lint.extend-select = [
 lint.isort = {known-first-party = ["helpers", "package_info", "pipx"]}
 lint.mccabe.max-complexity = 15
 
+[tool.codespell]
+# Ref: https://github.com/codespell-project/codespell#using-a-config-file
+skip = '.git,*.pdf,*.svg,.nox'
+check-hidden = true
+# case sensitive etc
+ignore-regex = '\b(UE|path/doesnt/exist)\b'
+ignore-words-list = 'astroid'
```
Comment (chrysle) [inline pyproject.toml:80]: I guess we don't need this, yet?

```suggestion
```

**Review by chrysle — COMMENTED** [review]
Still waiting for `codespell` to support `.gitignore` files.

File: pyproject.toml (hunk 3 above)
Comment (yarikoptic) [reply pyproject.toml:80]:
```
❯ codespell
./testdata/tests_packages/macos21-python3.12.txt:18: astroid ==> asteroid
./testdata/tests_packages/macos21-python3.12.txt:19: astroid ==> asteroid
./testdata/tests_packages/unix-python3.10.txt:18: astroid ==> asteroid
./testdata/tests_packages/unix-python3.10.txt:19: astroid ==> asteroid
./testdata/tests_packages/unix-python3.11.txt:18: astroid ==> asteroid
./testdata/tests_packages/unix-python3.11.txt:19: astroid ==> asteroid
./testdata/tests_packages/unix-python3.12.txt:18: astroid ==> asteroid
./testdata/tests_packages/unix-python3.12.txt:19: astroid ==> asteroid
./testdata/tests_packages/unix-python3.8.txt:18: astroid ==> asteroid
./testdata/tests_packages/unix-python3.8.txt:19: astroid ==> asteroid
./testdata/tests_packages/unix-python3.9.txt:18: astroid ==> asteroid
./testdata/tests_packages/unix-python3.9.txt:19: astroid ==> asteroid
./testdata/tests_packages/win-python3.12.txt:18: astroid ==> asteroid
./testdata/tests_packages/win-python3.12.txt:19: astroid ==> asteroid
```

sure thing we could just ignore the `testdata`. What would you prefer?

Comment (chrysle) [reply pyproject.toml:80]: Let's do the latter. I've edited the suggestion accordingly.

Comment (yarikoptic) [reply pyproject.toml:80]: thanks! just note that `skip` is already set above to some other globs -- I think this latter would override it. Please suggest on that line , or just let me know - I can do that

File: pyproject.toml [hunk 4]
```diff
@@ -71,6 +71,14 @@ lint.extend-select = [
 lint.isort = {known-first-party = ["helpers", "package_info", "pipx"]}
 lint.mccabe.max-complexity = 15
 
+[tool.codespell]
+# Ref: https://github.com/codespell-project/codespell#using-a-config-file
+skip = '.git,*.pdf,*.svg,.nox'
```
Comment (chrysle) [inline pyproject.toml:76]:
```suggestion
skip = '.git,*.pdf,*.svg,.nox,testdata'
```

Whoops, sorry for the extra trouble.

File: pyproject.toml (hunk 3 above)
Comment (yarikoptic) [reply pyproject.toml]: done

File: pyproject.toml (hunk 4 above)
Comment (yarikoptic) [reply pyproject.toml]: oops, missed this, did myself too. Thanks for the suggestion, and no trouble ;)

Comment (chrysle) [reply pyproject.toml]: Let's do it, then.

**Review by chrysle — APPROVED** [review]

**Comment by chrysle** [comment]
@yarikoptic Could you rebase again, please?

**Comment by yarikoptic** [comment]
done. Also added to skip `.mypy_cache`

**Review by chrysle — APPROVED** [review]

**Pull request merged**